src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3116,32 +3152,67 @@ private Message GetSortedSetCombineAndStoreCommandMessage(SetOperation operation
             {
                 SetOperation.Intersect => RedisCommand.ZINTERSTORE,
                 SetOperation.Union => RedisCommand.ZUNIONSTORE,
+                SetOperation.Difference => RedisCommand.ZDIFFSTORE,
                 _ => throw new ArgumentOutOfRangeException(nameof(operation)),
             };
             if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFFSTORE cannot be used with weights or with aggregation.");
 
-            List<RedisValue>? values = null;
+            List<RedisValue>? values = new List<RedisValue>();
```

Comment:
We shouldn't new up the list here if we're not going to use it - should be the previous approach. The ref arg should be nullable co compensate so the reference is set coming back, though.